mm2src/mm2_main/src/rpc/lp_commands/lp_commands_legacy.rs

```diff
@@ -70,50 +73,47 @@ pub async fn disable_coin(ctx: MmArc, req: Json) -> Result<Response<Vec<u8>>, St
         Ok(None) => return ERR!("No such coin: {}", ticker),
         Err(err) => return ERR!("!lp_coinfind({}): ", err),
     };
-    let coins_ctx = try_s!(CoinsContext::from_ctx(&ctx));
-
-    // If a platform coin is to be disabled, we get all the enabled tokens for this platform coin first.
-    let mut tokens_to_disable = coins_ctx.get_tokens_to_disable(&ticker).await;
-    // We then add the platform coin to the list of the coins to be disabled.
-    tokens_to_disable.insert(ticker.clone());
 
     // Get all matching orders and active swaps.
-    let active_swaps = try_s!(active_swaps_using_coins(&ctx, &tokens_to_disable));
-    let still_matching_orders = try_s!(get_matching_orders(&ctx, &tokens_to_disable).await);
+    let coins_to_disable: HashSet<_> = std::iter::once(ticker.clone()).collect();
+    let active_swaps = try_s!(active_swaps_using_coins(&ctx, &coins_to_disable));
+    let still_matching_orders = try_s!(get_matching_orders(&ctx, &coins_to_disable).await);
 
     // If there're matching orders or active swaps we return an error.
     if !active_swaps.is_empty() || !still_matching_orders.is_empty() {
-        let err = String::from("There're currently matching orders or active swaps for some tokens");
-        return disable_coin_err(err, &still_matching_orders, &[], &active_swaps);
+        let err = "There are currently matching orders or active swaps for some tokens".to_string();
+        return disable_coin_err(err, &still_matching_orders, &[], &active_swaps, &[]);
+    }
+
+    let coins_ctx = try_s!(CoinsContext::from_ctx(&ctx));
+    // If a platform coin is to be disabled, we should check if there are any tokens dependent on it.
+    let dependent_tokens = coins_ctx.get_dependent_tokens(&ticker).await;
+    if !dependent_tokens.is_empty() {
+        let err = format!("There are tokens dependent on '{ticker}'");
+        return disable_coin_err(err, &still_matching_orders, &[], &active_swaps, &dependent_tokens);
```

Comment:
The 2 instances of `disable_coin_err` can be merged into one error return. Right now, if there are active swaps/still matching orders, to get the dependant tokens the user will have to call this RPC again after the active swaps/still matching orders finish. Again, I am not sure what would be prefered for this from the GUI side as there might be new tokens that got enabled before the active swaps/still matching orders finish.
cc @smk762 @ca333

Reply:
For individual coins, a warning is show if trying to disable, and the disable action is either blocked or requires confirmation.

I think the gui side should be like:
- I have BNB, and a few BEP20 tokens active. I have a BUSD-BEP20 swap in progress.
- Try to disable BNB  ----> block action and show warning reason. Give option to disable all BEP20 tokens except BUSD-BEP20.

In this event, GUI could do a [legacy disable](https://developers.komodoplatform.com/basic-docs/atomicdex-api-legacy/disable_coin.html) batch request for the other coins. To simplify this, the API should return `coins_with_swaps` in response to exclude them from the batch disable.

Reply:
Such option was removed as per this requirement https://github.com/KomodoPlatform/atomicDEX-API/pull/1649#pullrequestreview-1293081247
> Give option to disable all BEP20 tokens except BUSD-BEP20.

Now we simply return an error if there are dependent tokens.

Reply:
Done